### ISSUE TITLE
Add LSTM forecasting model to ml_models

### ISSUE THREAD

**Issue by sufirehman** [open-energy-transition/demandcast#133]
### Summary

**Is your feature request related to a problem? Please describe.**

The README's planned features list mentions adding new ML models for forecasting (e.g. timesfm), and currently `ml_models/` only contains a single model implementation (`xgboost.py`). Having a second, fundamentally different model family (a neural sequence model rather than a gradient-boosted tree model) would let users compare approaches and would make the existing `if algorithm.lower() == "xgboost":` dispatch pattern in `train.py` and `forecast.py` genuinely extensible rather than single-purpose.

**Describe the solution you'd like**

I'd like to add `ml_models/lstm.py`, an LSTM-based forecasting model following the exact same interface as `xgboost.py`: `get_initialized_model()`, `train(prepared_dataset)`, `predict(model, prepared_dataset)`, `load(model_path)`, and `save(model, model_name)`, reading its hyperparameters from a corresponding `lstm_config.yaml` validated via Pydantic, matching the existing configuration pattern.

This would be backed by a peer-reviewed publication I authored comparing LSTM, Seq2Seq, and Prophet for electricity demand forecasting on UK smart-meter data (published in Springer Energy Informatics), so the implementation would draw on a tested, published methodology rather than a first attempt.

I'd add the new `elif algorithm.lower() == "lstm":` branches to `train.py` and `forecast.py` alongside the existing XGBoost branch, with no changes to existing XGBoost behaviour, plus unit tests in `tests/test_ml.py` following the existing mocking pattern.

**Describe alternatives you've considered**

Could start with Seq2Seq or a Prophet wrapper instead, but LSTM is the most natural first addition since it's the most directly comparable architecture to test against the existing XGBoost baseline using the same prepared_dataset structure.

**Additional context**

Happy to open a draft PR if this approach sounds right, or adjust based on any feedback on the interface or config structure first.

### Example

Once added, training would look exactly like it does today for XGBoost, just with a different algorithm name in the config:

algorithm: lstm

Running `train.py` would then dispatch to `ml_models.lstm.train(prepared_dataset)` instead of the XGBoost branch, using the same prepared_dataset structure already produced by the existing data pipeline. No change to the CLI, no change to how data is retrieved or prepared, just a new model option available wherever `algorithm` is set in the configuration.

For example, a user currently doing:

python train.py --config xgboost_config.yaml

would be able to do:

python train.py --config lstm_config.yaml

and get back a trained LSTM model saved and loaded the same way the XGBoost model is today, with predictions returned in the same pandas Series / dict-of-Series format `predict()` already returns, so anything downstream (evaluation, plotting, forecast.py) keeps working without modification.

**Comment by eantonini** [comment]
Thanks for opening this issue and for the detailed proposal.

In principle, we are open to adding additional ML models to DemandCast. Our preference is to include models that are widely used, reasonably established in the demand-forecasting literature, and maintainable within the scope of the project. An LSTM model seems like a reasonable candidate from that perspective, especially if the implementation builds on a tested methodology rather than being an experimental one-off.

That said, I would prefer to treat this carefully before merging it as a supported model. DemandCast’s current use case is not only short-term forecasting on a single smart-meter dataset, but global/scenario-driven hourly demand forecasting across countries and subdivisions, with heterogeneous data availability and exogenous weather/socioeconomic features. So the key question for us is not only whether LSTM works in general, but whether it adds value within the DemandCast workflow.

A draft PR would be welcome, provided it keeps the current framework unchanged and includes:
- the same public model interface as the existing `xgboost.py` implementation;
- a validated `lstm_config.yaml` following the existing configuration pattern;
- unit tests consistent with the current testing approach;
- clear documentation of the model assumptions, required dependencies, and expected input/output format;
- a small benchmark against the existing XGBoost model on one or more existing DemandCast datasets, using the same train/validation/test logic and evaluation metrics;
- reasonable defaults so that the model can run reproducibly without requiring excessive training time or hardware.

I would also suggest keeping the LSTM as an optional model at this stage. So overall: yes, this sounds potentially aligned with the roadmap, and I would be happy to review a draft PR if it is framed as an optional, benchmarked model implementation.

**Comment by sufirehman** [comment]
Hi @eantonini, I've opened the draft PR as discussed, all six items from your checklist are now complete, including the XGBoost-vs-LSTM benchmark on real DemandCast data (DNK, AUT, PRT, 2021-2023).

The benchmark result is honest: XGBoost outperforms LSTM on this small dataset (mean test MAPE 9.6% vs 16.5%), which is consistent with the underfitting expected when training a sequence model on only three entities. The full analysis and explanation is in the updated ML.md. Happy to discuss the results or adjust anything before you review.

PR: #135
